lib/globalize/active_record/adapter_dirty.rb

```diff
@@ -41,7 +41,8 @@ def clear_dirty
       end
 
       def _reset_attribute name
-        record.send(:restore_attribute!, name) if record.respond_to? :restore_attribute!, true
+        record.send("#{name}=", record.changed_attributes[name])
+        record.send(:attributes_changed_by_setter).send(:except!, name)
```

Comment:
This fixes the issue but is not what I want to keep as a solution since this is very much a private method we are calling. But at least this is a start.